iocore/dns/DNS.cc

```diff
@@ -784,11 +856,13 @@ DNSHandler::recv_dns(int /* event ATS_UNUSED */, Event * /* e ATS_UNUSED */)
                 ats_ip_ntop(&dnsc->ip.sa, ipbuff2, sizeof ipbuff2));
         continue;
       }
+      buf              = hostent_cache;
       hostent_cache    = nullptr;
       buf->packet_size = res;
       Debug("dns", "received packet size = %d", res);
+    Lsuccess:
       if (dns_ns_rr) {
-        Debug("dns", "round-robin: nameserver %d DNS response code = %d", dnsc->num, get_rcode(buf));
+        Debug("dns", "round-robin: nameserver %d DNS response code = %d", dnsc->num, get_rcode(buf.get()));
```

Comment:
This way keeps consistency with the old code. You're right. No point to do that.
And since this is the only reference to `get_rcode(HostEnt *ent)`.
I removed the function.Besides, the function itself is wrong implemented. It should be `get_rcode(reinterpret_cast<char *>(ent->buf))` instead of `get_rcode(reinterpret_cast<char *>(ent))`.